EDU-1087: add note about tctl deprecation and link CLI release notes (#2312)
temporalio/documentation: docs-src/concepts/what-is-the-temporal-cli.md

  - cli
---

:::note

> When upgrading from `tctl` to CLI, make sure to update your environment variables and use updated commands.

Reviewer: ```suggestion
When upgrading from `tctl` to Temporal CLI, make sure to update your environment variables and use updated commands.
```

   - [Windows amd64](https://temporal.download/cli/archive/latest?platform=windows&arch=amd64)
   - [Windows arm64](https://temporal.download/cli/archive/latest?platform=windows&arch=arm64)
2. Extract the downloaded archive.
3. Add the `temporal` binary to your PATH (use `temporal.exe` for Windows).

## Starting the Temporal Server

> To start the Temporal Server and Web UI, run the following command:

Reviewer 2: ```suggestion
To start the Temporal Server, run the following command:
```

```bash
temporal server start-dev
```

> This launches a server on `localhost:7233` and a web interface at <http://localhost:8233>.

Reviewer: ```suggestion
The command launches a server on `localhost:7233` and a web interface at <http://localhost:8233>.
```

By default, data isn't persisted.

> If you want to save Workflows, use the `--db-filename` flag:

Reviewer: ```suggestion
If you want to save Workflows, use the `--db-filename` option:
```

]
```

> Filter out Workflows based on Type with [jq](https://stedolan.github.io/jq/):

Reviewer: ```suggestion
Filter out Workflows based on Workflow Type with [jq](https://stedolan.github.io/jq/):
```
Or can simply lowercase "type" in this instance. Either way is good.

   2 "MyWorkflow"
```

> To see the full range of Workflow-related commands, run `temporal workflow` or visit [CLI ▶️ workflow](/cli/workflow).

Reviewer: ```suggestion
To see the full range of Workflow-related commands, run `temporal workflow` or see the [Temporal CLI workflow command reference](/cli/workflow).
```
Per https://developers.google.com/style/cross-references#references-to-other-documents.

> For a full list of available commands, run `temporal` or visit [Available commands](#available-commands).

Reviewer: ```suggestion
For a full list of available commands, run `temporal` or see [Available commands](#available-commands).
```

```

## Server configuration

View the full list of CLI configuration options with the `--help` flag:

```bash
temporal server start-dev --help
```

### Namespace registration

Namespaces are pre-registered at startup for immediate use.
Customize pre-registered Namespaces with the following command:

```bash
temporal server start-dev --namespace foo --namespace bar
```

Register Namespaces with `namespace create`:

```bash
temporal operator namespace create foo
```

### Enable or disable Temporal UI

By default, the Temporal UI starts with Temporal CLI.
Disable the UI with:

```bash
temporal server start-dev --headless
```

### Dynamic configuration

Advanced Temporal CLI configuration requires a dynamic configuration file.

To set values on the command line, use `--dynamic-config-value KEY=JSON_VALUE`.
For example, enable the Search Attribute cache:

```bash
temporal server start-dev --dynamic-config-value system.forceSearchAttributesCacheRefreshOnRead=false
```

This setting makes created Search Attributes immediately available.

## Environmental variables

Configure system environment with Temporal CLI environmental variables.

> Use the table below as reference:

Reviewer 2: ```suggestion
The following table lists and describes the environment variables you can set for Temporal CLI.
```

source <(temporal completion zsh)
```

> If you're running auto-completion from the terminal, run:

Reviewer: ```suggestion
If you're running auto-completion from the terminal, run the following command:
```
Per https://developers.google.com/style/colons#introductory-phrase-preceding-colon.

```sh
echo 'source <(temporal completion zsh)' >> ~/.zshrc
```

> After setting the variable, run:

Reviewer: ```suggestion
After setting the variable, run the following command:
```

`source ~/.zshrc`.

### Bash auto-completion

> Install []`bash-completion`](https://github.com/scop/bash-completion#installation) and add the software to your `~/.bash_profile`.

Reviewer: ```suggestion
Install [bash-completion](https://github.com/scop/bash-completion#installation) and add the software to `~/.bash_profile`.
```

> To utilize aliases, add them to `~/.bash_profile` as needed.

Reviewer: ```suggestion
To use aliases, add them to `~/.bash_profile` as needed.
```
